docs/deep-dives/esm.md

```diff
@@ -255,7 +259,52 @@ module.exports = {
 
 [`'...'` syntax]: https://webpack.js.org/configuration/resolve/#resolveconditionnames
 
-### 3. Switch to Vitest and ESM
+### 3. Switch to tsdown
+
+Within `build-package` we will switch to calling [`tsdown`](https://tsdown.dev/) instead of wrapping the TypeScript compiler directly. This will allow us to continue publishing both CJS and ESM after we've migrated to ESM.
+
+#### Create tsdown.config.mjs
+
+```javascript
+import { defineConfig } from 'tsdown/config';
+
+export default defineConfig({
+  entry: ['src/index.ts'],
+  format: ['esm', 'cjs'],
+  exports: {
+    devExports: '@seek/my-repo/source',
+  },
+  // TODO: Consider removing this if your package can be bundled
+  unbundle: true,
+});
+```
+
+#### Migrate `assets` usage
+
+Skuba currently supports copying additional asset files during the `build-package` command via the `assets` field in `skuba.assets` within `package.json`. As `tsdown` does not support this field, we will need to migrate these assets to use the `copy` field in `tsdown.config.mjs` instead.
+
+```json
+{
+  "skuba": {
+    "assets": ["**/*.vocab/*translations.json"]
+  }
+}
+```
+
+```diff
+export default defineConfig({
+  entry: ['src/index.ts'],
+  format: ['esm', 'cjs'],
+  exports: {
```

Comment:
Setting `exports` automagically manages your package.json files. Do we want to consider returning a failing status code when running build-package and `package.json` is modified so that consumers know to commit the file?